ComplementaryScripts/missingFields/mapIDsViaMNXref.m

```diff
@@ -0,0 +1,167 @@
+function targetList=mapIDsViaMNXref(type,queryList,fromDB,toDB)
+% mapIDsViaMNXref
+%
+%   Associate reaction/metabolite identifiers between different databases
+%
+%   type            'rxns' or 'mets' depending on which information is
+%                   expected to associate
+%   queryList       cell array of reaction/metabolite in the query database
+%                   (e.g. model.rxnKEGGID, model.metChEBIID)
+%   fromDB          the query database name
+%   toDB            the subject database name
+%
+%   targetList      cell array of reaction/metabolite in the subjuct database
+%
+%   Note: This function map metabolite identifiers across
+%   different databases using the MNXref naming convention
+%   reference MNX structure comes from RAVEN repository
+%
+%   Usage: targetList=mapIDsViaMNXref(type,queryList,fromDB,toDB)
+%
+%   Hao Wang, 2018-05-23
+%
+
+targetList={};
+
+if nargin<4
+		EM='Missing input arguments';
+		disp(EM);
+end
+
+if isequal(fromDB,toDB)
+		EM='Query and subject databases cannot be the same!';
+		disp(EM);
+end
+
+% Load MNXref data structure, this part of code is modified by Feiran, in
+% order to load data structure from RAVEN repository since the data is
+% stored in raven toolbox.
+%since right now, MNX reference structure only exist in the RAVEN branch:
+%origin/feat/add_MetaNetX, we need to switch devel branch to that branch, load
+%the file and switch back.
+%remember when this branch is merged to the master branch, change codes
+%below.
+IDfile = 'ravenCobraWrapper.m';
+currentPath = pwd;
+%Try to find root of toolbox:
+try
+    toolboxPath = which(IDfile);                %full file path
+    %slashPos    = getSlashPos(toolboxPath);
+     slashPos = strfind(toolboxPath,'/'); 
+    toolboxPath = toolboxPath(1:slashPos(end)); %folder path
+    %Go up until the root is found:
+     D = dir(toolboxPath);    
+    while ~ismember({'.git'},{D.name})
+        %slashPos    = getSlashPos(toolboxPath);
+         slashPos = strfind(toolboxPath,'/'); 
+        toolboxPath = toolboxPath(1:slashPos(end-1));
+        D = dir(toolboxPath);
+    end
+    cd(toolboxPath);
+    
+catch
+    disp([toolbox ' toolbox cannot be found'])
+end
+%Check if this branch is feat/add_MetaNetX
+    currentBranch = git('rev-parse --abbrev-ref HEAD');
+    if ~strcmp(currentBranch,'feat/add_MetaNetX')
+            git checkout -B feat/add_MetaNetX origin/feat/add_MetaNetX
+            currentBranch = git('rev-parse --abbrev-ref HEAD');

```

Comment:
your code produces the error `fatal: Ambiguous object name: 'origin/feat/add_MetaNetX'` in my computer. `git checkout feat/add_MetaNetX` will work regardless if the branch is stored locally or not; in the later case it will just download it from the remote. If you want I can push this simplification to this branch :)

Reply:
I tried yesterday it worked in my computer, but I have already simplified it in the new commit.